src/ripple/app/tx/impl/AMMDeposit.cpp

```diff
@@ -63,17 +63,20 @@ AMMDeposit::preflight(PreflightContext const& ctx)
     }
     else if (flags & tfLPToken)
     {
-        if (!lpTokens || amount || amount2 || ePrice)
+        // if included then both amount and amount2 are deposit min
+        if (!lpTokens || ePrice || (amount && !amount2) || (!amount && amount2))
             return temMALFORMED;
     }
     else if (flags & tfSingleAsset)
     {
-        if (!amount || lpTokens || amount2 || ePrice)
+        // if included then lpTokens is deposit min
+        if (!amount || amount2 || ePrice)
             return temMALFORMED;
     }
     else if (flags & tfTwoAsset)
     {
-        if (!amount || !amount2 || lpTokens || ePrice)
+        // if included then lpTokens is deposit min
+        if (!amount || !amount2 || ePrice)
             return temMALFORMED;
     }
```

Comment:
I am not so sure how I feel about fields having different meanings depending on flag values. It's the max if this, min if that... OTOH, I know adding more fields is an annoyance.

Reply:
We already have a precedent for different meanings though within the same flag. Deposit options 3 (tfTwoAsset) and 5 (tfLimitLPToken) already use `amount` and `amount2` as either the actual deposit amount or as the maximum deposit amount.